server/lib/actors/hot.go

```diff
@@ -0,0 +1,741 @@
+package actors
+
+import (
+	"log"
+	"fmt"
+	"time"
+	"sort"
+	"sync"
+	"math"
+	"sync/atomic"
+
+	proto "github.com/umegaya/channer/server/proto"
+	"github.com/umegaya/yue"
+	"github.com/umegaya/channer/server/lib/utils"
+	yuep "github.com/umegaya/yue/proto"
+)
+
+const (
+	INVALID_HOTBUCKET = iota
+	SUMMERY_HOUR_HOTBUCKET
+	SUMMERY_DAY_HOTBUCKET
+	SUMMERY_WEEK_HOTBUCKET
+	SUMMERY_ALLTIME_HOTBUCKET
+)
+
+const (
+	DEFAULT_HOTBUCKET_SIZE = 1024 * 10
+	PER_CHANNEL_HOTBUCKET_SIZE = 64
+	ALLTIME_SUMMERY_LIMIT = 10000
+	GOAT_UPDATE_SPAN_SEC = 60 * 60 * 12 //12h
+)
+
+//
+// FetchResult 
+//
+// FetchResult represents data format which fetcher should return
+type FetchResult struct {
+	id proto.UUID		//target id
+	parent proto.UUID	//target parent id. FetchResult will be group by parent
+	score int32			//target score. higher score regard as "hot".
+	vote uint32 		//number of vote (up + down)
+}
+
+
+//
+// entry and its sorter
+//
+// HotEntry represents single hot object with its score and id
+type HotEntry struct {
+	Id proto.UUID
+	Score int32
+	Vote uint32
+}
+
+func (e *HotEntry) within(start, end time.Time) bool {
+	return yue.UUIDBetween(yuep.UUID(e.Id), start, end)
```

Comment:
__[gometalinter]__
* [gotype] undeclared name: yue
* [gotype] cannot convert e.Id (variable with invalid type) to github.com/umegaya/yue/proto.UUID